### ISSUE TITLE
scalatra-swagger fails with json4s 3.2.11

### ISSUE THREAD

**Issue by okapies** [scalatra/scalatra#446]
It can be reproduced using the official [swagger-example](https://github.com/scalatra/scalatra-website-examples/blob/master/2.3/swagger-example/project/build.scala) and just modifying as the following:

```
"org.json4s"   %% "json4s-native" % "3.2.11",
```

And start container to access *api-docs*.

```
$ ./sbt container:start
...
$ curl http://localhost:8080/api-docs
```

**Comment by futurechimp** [comment]
Hi @okapies,

Thanks for the bug report. You're right, the example Swagger app fails with json4s 3.2.11. The latest I can get it to work with is json4s 3.2.10.

As a workaround, you might consider dropping back to json4s 3.2.10. Alternately, the Scalatra 2.4.0-SNAPSHOT works, although it may bring other problems.

@casualjim or @rossabaker can you see a reason why json4s 3.2.11 won't work with Scalatra 2.3.0's Swagger implementation? It seems to die here, in the type hinting:

https://github.com/json4s/json4s/blob/v3.2.11/core/src/main/scala/org/json4s/Formats.scala#L324

Here's the stack trace:

```
java.lang.AbstractMethodError
	at org.json4s.DefaultFormats$class.$init$(Formats.scala:324)
	at org.scalatra.swagger.SwaggerSerializers$$anon$10.<init>(SwaggerSerializers.scala:149)
	at org.scalatra.swagger.SwaggerSerializers$.<init>(SwaggerSerializers.scala:149)
	at org.scalatra.swagger.SwaggerSerializers$.<clinit>(SwaggerSerializers.scala)
	at org.scalatra.swagger.SwaggerBase$class.jsonFormats(SwaggerBase.scala:102)
	at org.scalatra.example.swagger.ResourcesApp.jsonFormats(FlowersSwagger.scala:7)
	at org.scalatra.swagger.SwaggerBaseBase$$anonfun$renderIndex$7.apply(SwaggerBase.scala:89)
	at org.scalatra.swagger.SwaggerBaseBase$$anonfun$renderIndex$7.apply(SwaggerBase.scala:89)
	at scala.Option.map(Option.scala:145)
	at org.scalatra.swagger.SwaggerBaseBase$class.renderIndex(SwaggerBase.scala:89)
	at org.scalatra.example.swagger.ResourcesApp.renderIndex(FlowersSwagger.scala:7)
	at org.scalatra.swagger.SwaggerBaseBase$$anonfun$initialize$3.apply(SwaggerBase.scala:47)
	at org.scalatra.swagger.SwaggerBaseBase$$anonfun$initialize$3.apply(SwaggerBase.scala:47)
	at org.scalatra.ScalatraBase$class.org$scalatra$ScalatraBase$$liftAction(ScalatraBase.scala:270)
	at org.scalatra.ScalatraBase$$anonfun$invoke$1.apply(ScalatraBase.scala:265)
	at org.scalatra.ScalatraBase$$anonfun$invoke$1.apply(ScalatraBase.scala:265)
	at org.scalatra.ApiFormats$class.withRouteMultiParams(ApiFormats.scala:178)
	at org.scalatra.example.swagger.ResourcesApp.withRouteMultiParams(FlowersSwagger.scala:7)
	at org.scalatra.ScalatraBase$class.invoke(ScalatraBase.scala:264)
	at org.scalatra.example.swagger.ResourcesApp.org$scalatra$json$JsonSupport$$super$invoke(FlowersSwagger.scala:7)
	at org.scalatra.json.JsonSupport$$anonfun$invoke$1.apply(JsonSupport.scala:74)
	at org.scalatra.json.JsonSupport$$anonfun$invoke$1.apply(JsonSupport.scala:68)
	at org.scalatra.ApiFormats$class.withRouteMultiParams(ApiFormats.scala:178)
	at org.scalatra.example.swagger.ResourcesApp.withRouteMultiParams(FlowersSwagger.scala:7)
	at org.scalatra.json.JsonSupport$class.invoke(JsonSupport.scala:68)
	at org.scalatra.example.swagger.ResourcesApp.invoke(FlowersSwagger.scala:7)
	at org.scalatra.ScalatraBase$$anonfun$runRoutes$1$$anonfun$apply$8.apply(ScalatraBase.scala:240)
	at org.scalatra.ScalatraBase$$anonfun$runRoutes$1$$anonfun$apply$8.apply(ScalatraBase.scala:238)
	at scala.Option.flatMap(Option.scala:170)
	at org.scalatra.ScalatraBase$$anonfun$runRoutes$1.apply(ScalatraBase.scala:238)
	at org.scalatra.ScalatraBase$$anonfun$runRoutes$1.apply(ScalatraBase.scala:237)
	at scala.collection.immutable.Stream.flatMap(Stream.scala:450)
	at org.scalatra.ScalatraBase$class.runRoutes(ScalatraBase.scala:237)
	at org.scalatra.ScalatraServlet.runRoutes(ScalatraServlet.scala:49)
	at org.scalatra.ScalatraBase$class.runActions$1(ScalatraBase.scala:163)
	at org.scalatra.ScalatraBase$$anonfun$executeRoutes$1.apply$mcV$sp(ScalatraBase.scala:175)
	at org.scalatra.ScalatraBase$$anonfun$executeRoutes$1.apply(ScalatraBase.scala:175)
	at org.scalatra.ScalatraBase$$anonfun$executeRoutes$1.apply(ScalatraBase.scala:175)
	at org.scalatra.ScalatraBase$class.org$scalatra$ScalatraBase$$cradleHalt(ScalatraBase.scala:193)
	at org.scalatra.ScalatraBase$class.executeRoutes(ScalatraBase.scala:175)
	at org.scalatra.ScalatraServlet.executeRoutes(ScalatraServlet.scala:49)
	at org.scalatra.ScalatraBase$$anonfun$handle$1.apply$mcV$sp(ScalatraBase.scala:113)
	at org.scalatra.ScalatraBase$$anonfun$handle$1.apply(ScalatraBase.scala:113)
	at org.scalatra.ScalatraBase$$anonfun$handle$1.apply(ScalatraBase.scala:113)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:58)
	at org.scalatra.DynamicScope$class.withResponse(DynamicScope.scala:80)
	at org.scalatra.ScalatraServlet.withResponse(ScalatraServlet.scala:49)
	at org.scalatra.DynamicScope$$anonfun$withRequestResponse$1.apply(DynamicScope.scala:60)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:58)
	at org.scalatra.DynamicScope$class.withRequest(DynamicScope.scala:71)
	at org.scalatra.ScalatraServlet.withRequest(ScalatraServlet.scala:49)
	at org.scalatra.DynamicScope$class.withRequestResponse(DynamicScope.scala:59)
	at org.scalatra.ScalatraServlet.withRequestResponse(ScalatraServlet.scala:49)
	at org.scalatra.ScalatraBase$class.handle(ScalatraBase.scala:111)
	at org.scalatra.ScalatraServlet.org$scalatra$servlet$ServletBase$$super$handle(ScalatraServlet.scala:49)
	at org.scalatra.servlet.ServletBase$class.handle(ServletBase.scala:43)
	at org.scalatra.example.swagger.ResourcesApp.org$scalatra$CorsSupport$$super$handle(FlowersSwagger.scala:7)
	at org.scalatra.CorsSupport$$anonfun$handle$1.apply$mcV$sp(CORSSupport.scala:202)
	at org.scalatra.CorsSupport$$anonfun$handle$1.apply(CORSSupport.scala:190)
	at org.scalatra.CorsSupport$$anonfun$handle$1.apply(CORSSupport.scala:190)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:58)
	at org.scalatra.DynamicScope$class.withResponse(DynamicScope.scala:80)
	at org.scalatra.ScalatraServlet.withResponse(ScalatraServlet.scala:49)
	at org.scalatra.DynamicScope$$anonfun$withRequestResponse$1.apply(DynamicScope.scala:60)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:58)
	at org.scalatra.DynamicScope$class.withRequest(DynamicScope.scala:71)
	at org.scalatra.ScalatraServlet.withRequest(ScalatraServlet.scala:49)
	at org.scalatra.DynamicScope$class.withRequestResponse(DynamicScope.scala:59)
	at org.scalatra.ScalatraServlet.withRequestResponse(ScalatraServlet.scala:49)
	at org.scalatra.CorsSupport$class.handle(CORSSupport.scala:188)
	at org.scalatra.example.swagger.ResourcesApp.handle(FlowersSwagger.scala:7)
	at org.scalatra.ScalatraServlet.service(ScalatraServlet.scala:54)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:738)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:551)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:568)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:221)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1111)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:478)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:183)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1045)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:199)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
	at org.eclipse.jetty.server.Server.handle(Server.java:462)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:279)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:232)
	at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:534)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:607)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:536)
	at java.lang.Thread.run(Thread.java:745)

```

**Comment by seratch** [comment]
The upcoming 2.3.1 release will fix this issue because 2.3.1 will be built with json4s 3.2.11.

**Comment by seratch** [comment]
scalatra-swagger 2.3.1 is out.
